src/SparkTestBase.sol

```diff
@@ -3,10 +3,110 @@ pragma solidity ^0.8.0;
 
 import 'aave-helpers/ProtocolV3TestBase.sol';
 
+import { GovHelpers } from 'aave-helpers/GovHelpers.sol';
+
+import { IPoolAddressesProviderRegistry } from 'aave-v3-core/contracts/interfaces/IPoolAddressesProviderRegistry.sol';
+import { IPoolAddressesProvider }         from 'aave-v3-core/contracts/interfaces/IPoolAddressesProvider.sol';
+import { IPool }                          from 'aave-v3-core/contracts/interfaces/IPool.sol';
+
 import { IDaiInterestRateStrategy }    from "./IDaiInterestRateStrategy.sol";
 import { IDaiJugInterestRateStrategy } from "./IDaiJugInterestRateStrategy.sol";
 
-contract SparkTestBase is ProtocolV3_0_1TestBase {
+abstract contract SparkTestBase is ProtocolV3TestBase {
+
+    address internal executor;
+    address internal payload;
+    
+    string internal domain;
+    string internal id;
+
+    IPoolAddressesProviderRegistry internal poolAddressesProviderRegistry;
+    IPoolAddressesProvider         internal poolAddressesProvider;
+    IPool                          internal pool;
+
+    function loadPoolContext(address poolProvider) internal {
+        poolAddressesProvider = IPoolAddressesProvider(poolProvider);
+        pool                  = IPool(poolAddressesProvider.getPool());
+    }
+
+    function deployPayload() internal returns (address) {
+        return deployCode(string(abi.encodePacked('Spark', domain, '_', id, '.sol')));
+    }
+
+    function testSpellExecutionDiff() public {
+        address[] memory poolProviders = poolAddressesProviderRegistry.getAddressesProvidersList();
+        string memory prefix = string(abi.encodePacked(id, '-', domain));
+
+        for (uint256 i = 0; i < poolProviders.length; i++) {
+            loadPoolContext(poolProviders[i]);
+
+            createConfigurationSnapshot(
+                string(abi.encodePacked(prefix, '-', vm.toString(address(pool)), '-pre')),
+                pool
+            );
+        }
+
+        GovHelpers.executePayload(vm, payload, executor);
+
+        for (uint256 i = 0; i < poolProviders.length; i++) {
+            loadPoolContext(poolProviders[i]);
+
+            createConfigurationSnapshot(
+                string(abi.encodePacked(prefix, '-', vm.toString(address(pool)), '-post')),
+                pool
+            );
+
+            diffReportsFixed(
+                string(abi.encodePacked(prefix, '-', vm.toString(address(pool)), '-pre')),
+                string(abi.encodePacked(prefix, '-', vm.toString(address(pool)), '-post'))
+            );
+        }
+    }
+
```

Comment:
Can we use this to move to `test_bytecodeMatches` test name syntax? Find it cleaner and its more widely used in foundry

Reply:
Function LGTM

Reply:
Probably should be called `test_payloadBytecodeMatches`, I thought this was for the implementation at first

Reply:
We should probably add a bytecode comparison against the AAVE 3.0.2 pool using the same logic no?

Reply:
This type of comparison only works for contracts that don't contain immutables, libraries, etc. I think it makes sense for the spell, but beyond that the method of local sourcify verification works better imo.